File: java/src/test/java/com/alxarafe/domain/catalogue/valueobject/SkuTest.java
package com.alxarafe.domain.catalogue.valueobject;

import com.alxarafe.app.domain.catalogue.valueobject.Sku;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class SkuTest {

    @Test
    void create() {
        var sku = new Sku("WIDGET-001");
        assertEquals("WIDGET-001", sku.value());
    }

    @Test
    void nullValue() {
        assertThrows(NullPointerException.class, () -> new Sku(null));
    }

    @Test
    void empty() {
        assertThrows(IllegalArgumentException.class, () -> new Sku(""));
    }

    @Test
    void blank() {
        assertThrows(IllegalArgumentException.class, () -> new Sku("   "));
    }

    @Test
    void tooLong() {
        assertThrows(IllegalArgumentException.class, () -> new Sku("A".repeat(51)));
    }

    @Test
    void testToString() {
        var sku = new Sku("TEST");
        assertEquals("TEST", sku.toString());
    }
}

File: php/tests/Unit/Domain/Catalogue/ValueObject/SkuTest.php
<?php

declare(strict_types=1);

namespace Tests\Unit\Domain\Catalogue\ValueObject;

use Alxarafe\App\Domain\Catalogue\ValueObject\Sku;
use InvalidArgumentException;
use PHPUnit\Framework\TestCase;

final class SkuTest extends TestCase
{
    public function testCreate(): void
    {
        $sku = new Sku('WIDGET-001');
        self::assertSame('WIDGET-001', $sku->value());
    }

    public function testEmpty(): void
    {
        $this->expectException(InvalidArgumentException::class);
        new Sku('');
    }

    public function testTooLong(): void
    {
        $this->expectException(InvalidArgumentException::class);
        new Sku(str_repeat('A', 51));
    }

    public function testEquals(): void
    {
        $a = new Sku('ABC');
        $b = new Sku('ABC');
        $c = new Sku('XYZ');
        self::assertTrue($a->equals($b));
        self::assertFalse($a->equals($c));
    }

    public function testToString(): void
    {
        $sku = new Sku('TEST');
        self::assertSame('TEST', (string) $sku);
    }
}
